call-home/src/bin/callhome/main.rs

```diff
@@ -172,58 +194,74 @@ async fn generate_report(
 
     let k8s_node_count = k8s_client.get_node_len().await;
     match k8s_node_count {
-        Ok(k8s_node_count) => report.k8s_node_count = k8s_node_count as u8,
+        Ok(k8s_node_count) => report.k8s_node_count = Some(k8s_node_count as u8),
         Err(err) => {
             error!("{:?}", err);
         }
     };
 
     // List of disks on each node.
     let mut node_disks = HashMap::new();
-    let nodes = http_client.nodes_api().get_nodes(None).await;
+    let nodes = Retry::spawn(retry_strategy.clone(), || async {
+        http_client.nodes_api().get_nodes(None).await
+    })
+    .await;
     match nodes {
         Ok(nodes) => {
             let nodes = nodes.into_body();
             for node in &nodes {
-                if let Ok(b_devs_result) = http_client
-                    .block_devices_api()
-                    .get_node_block_devices(&node.id, Some(true))
-                    .await
-                {
+                let b_devs_result = Retry::spawn(retry_strategy.clone(), || async {
+                    http_client
+                        .block_devices_api()
+                        .get_node_block_devices(&node.id, Some(true))
+                        .await
+                })
+                .await;
+                if let Ok(b_devs) = b_devs_result {
                     node_disks
                         .entry(node.id.to_string())
                         .or_insert_with(Vec::new)
-                        .extend(b_devs_result.into_body());
+                        .extend(b_devs.into_body());
                 };
             }
-            report.storage_node_count = nodes.len() as u8
+            report.storage_node_count = Some(nodes.len() as u8)
         }
         Err(err) => {
             error!("{:?}", err);
         }
     };
 
-    let pools = http_client.pools_api().get_pools(None).await;
+    let pools = Retry::spawn(retry_strategy.clone(), || async {
```

Comment:
I think this is not necessary, you can add a retry layer to the rest client itself through tower retry